cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -141,14 +141,14 @@ public void setTableRenderer(TableRenderer tableRenderer) {
 
   public void setConnectionConfig(@Nullable CLIConnectionConfig connectionConfig) {
     this.connectionConfig = connectionConfig;
-    clientConfig.setConnectionConfig(connectionConfig);
+    this.clientConfig = ClientConfig.builder().setConnectionConfig(connectionConfig).build();
```

Comment:
doesn't behavior change here? earlier, the state of the existing ``clientConfig`` was retained, but here it gets overwritten? Should pass ``clientConfig`` to the builder, no (at least when it is non-null)?